test/fx/test_dce_pass.py

```diff
@@ -233,63 +232,3 @@ def forward(self, a: torch.Tensor) -> torch.Tensor:
 
         # %add_out node should not be removed because it has side effects.
         self._run_dce_and_test(TestModule(), expect_dce_changes=False, custom=True)
-
-    @unittest.skipIf(IS_MACOS, "Not working on macos")
-    def test_keep_collectives(self):
-        """
-        Test that DCE doesn't remote collective ops even the results are not used.
-        """
-
-        from torch.testing._internal.distributed.fake_pg import FakeStore
-
-        class TestModule(torch.nn.Module):
-            def forward(
-                self, a: torch.Tensor, b: torch.Tensor, c: torch.Tensor
-            ) -> torch.Tensor:
-                d = torch.ops.aten.mul.Tensor(a, b)
-                e = torch.ops.aten.mul.Tensor(a, c)
-                future = torch.ops._c10d_functional.all_reduce.default(e, "sum", "0")
-                synced_e = torch.ops._c10d_functional.wait_tensor.default(
-                    future
-                )  # synced_e is not used
```

Comment:
Curious why do we need to remove these two tests in this file?

The motivation for this test case is that even if rank 0 doesn't use the `synced_e` (all-reduce output), rank 1 might use their `synced_e`, and hence we still cannot DCE the all_reduce and wait_tensor in rank 0

Reply:
This was testing fx tracing preservation in graph, relying on 
```
torch/distributed/_functional_collectives.py
-    # mark these ops has side effect so that they won't be removed by DCE
-    torch.fx.node.has_side_effect(torch.ops._c10d_functional.wait_tensor.default)
-    torch.fx.node.has_side_effect(torch.ops._c10d_functional.wait_tensor)
```
making wait_tensor effectful op will keep it from DCE for dynamo tracing only.

Potentially if fx case is needed - we can keep it.